lib/search-results-bar/index.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import IconButton from '../icon-button';
+import ChevronRightIcon from '../icons/chevron-right';
+
+import * as S from '../state';
+
+type StateProps = {
+  selectedSearchMatchIndex: number | null;
+  numberOfMatchesInNote: number;
+};
+
+type DispatchProps = {
+  setSearchSelection: (index: number) => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+const SearchResultsBar: FunctionComponent<Props> = ({
+  selectedSearchMatchIndex: index,
+  numberOfMatchesInNote: total,
+  setSearchSelection,
+}) => {
+  const nextButtonRef = useRef<HTMLButtonElement>();
+  const prevButtonRef = useRef<HTMLButtonElement>();
+
+  useEffect(() => {
+    const setPrev = (event: MouseEvent) => {
+      const newIndex = (total + (index ?? -1) - 1) % total;
+      setSearchSelection(newIndex);
+    };
+
+    const setNext = (event: MouseEvent) => {
+      const newIndex = (total + (index ?? -1) + 1) % total;
+      setSearchSelection(newIndex);
+    };
+    prevButtonRef.current?.addEventListener('click', setPrev, true);
+    nextButtonRef.current?.addEventListener('click', setNext, true);
+
+    return () => {
+      prevButtonRef.current?.removeEventListener('click', setPrev, true);
+      nextButtonRef.current?.removeEventListener('click', setNext, true);
+    };
+  }, [index, total]);
```

Comment:
Were there any issues in setting these directly on the `<span>`s with `onClick`?

```ts
<span className="search-results-next" onClick={() => setSearchSelection((total + (index ?? -1) - 1) % total)}>
```

Reply:
I swear I tried this and it didn't work but apparently I just didn't wire it up right 🤦